viewsb/packet.py

```diff
@@ -758,6 +758,108 @@ def from_setup_data(cls, setup_data, **fields):
+        elif descType == 1:
+            return 'device'
+        elif descType == 2:
+            return 'configuration'
+        elif descType == 3:
+            return 'string'
+        elif descType == 4:
+            return 'interface'
+        elif descType == 5:
+            return 'endpoint'
+        elif descType == 6:
+            return 'device qualifier'
+        elif descType == 7:
+            return 'other speed'
+        elif descType == 8:
+            return 'interface power'
+        elif descType == 9:
+            return 'otg'
+        elif descType == 10:
+            return 'debug'
+        elif descType == 11:
+            return 'interface association'
+        elif descType == 12:
+            return 'security'
+        elif descType == 13:
+            return 'key'
+        elif descType == 14:
+            return 'encryption type'
+
+        elif descType == 16:
+            return 'device capability'
+        elif descType == 17:
+            return 'wireless endpoint'
+
+        elif descType == 33:
+            return 'hid'
+        elif descType == 34:
+            return 'hid report'
+        elif descType == 35:
+            return 'physical description'
+        return 'unknown'
+
+    @staticmethod
+    def _decode_descriptor_lang(lang):
+        if lang == 0x0904:
+            return 'english'
+        return f'unknown {lang:#06x}'
+
+    @classmethod
+    def _decode_descriptor_request(cls, data):
+        index = data.wValue & 0xFF
+        descType = data.wValue >> 8
+        data.wValue = {
+            'Index': index,
+            'Type': cls._decode_descriptor_type(descType),
+        }
+        if data.wIndex != 0:
+            data.wIndex = {
+                'Language': cls._decode_descriptor_lang(data.wIndex),
+            }
+
+    @staticmethod
+    def _decode_request_type(reqType):
+        if reqType == 0:
+            return 'standard'
+        elif reqType == 1:
+            return 'class'
+        elif reqType == 2:
+            return 'vendor'
+        return 'reserved'
+
+    @staticmethod
+    def _decode_request_recipient(recipient):
+        if recipient == 0:
+            return 'device'
+        elif recipient == 1:
+            return 'interface'
+        elif recipient == 2:
+            return 'endpoint'
+        elif recipient == 3:
+            return 'other'
+        return 'reserved'
+
+    @classmethod
+    def _decode_request(cls, data):
+        if data.bRequest == 6:
+            cls._decode_descriptor_request(data)
+
+        reqDir = (data.bmRequestType >> 7)
+        reqType = (data.bmRequestType >> 5) & 0x3
+        reqRecipient = data.bmRequestType & 0x1F
+        data.bmRequestType = {
+            'Direction': 'device-to-host' if reqDir == 1 else 'host-to-device',
+            'Type': cls._decode_request_type(reqType),
+            'Recipient': cls._decode_request_recipient(reqRecipient),
+        }
+
+        data.bRequest = cls.REQUESTS.get(data.bRequest, 'UNKNOWN')
```

Comment:
Instead of just UNKNOWN, at least including the hex value would be nice.

Probably the same for "descriptor type" above.

Reply:
This should now be fixed in the latest commit